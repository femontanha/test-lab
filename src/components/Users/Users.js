import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './users.css';

const propTypes = {
  users: PropTypes.object.isRequired,
};

class Users extends PureComponent {
  renderUsers() {
    const { users } = this.props;
    return Object.keys(users).map((item) => (
      <ul className={ styles.list } key={ users[item].id }>
        <li>{ users[item].username }</li>
        <li>{ users[item].name }</li>
        <li>{ users[item].email }</li>
        <li>{ users[item].phone }</li>
        <li>{ users[item].website }</li>
      </ul>
    ));
  }

  render() {
    return (
      <div className={ styles.users }>
        <Link className="app-button" to="/">View Posts</Link>
        { this.renderUsers() }
      </div>
    );
  }
}

Users.propTypes = propTypes;

export default Users;
