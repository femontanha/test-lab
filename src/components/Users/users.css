.users {
  margin: 0 auto;
  max-width: 900px;
  padding: 30px;
  text-align: center;
}

.list {
  list-style: none;
  padding: 15px;
  max-width: 300px;
  margin: 30px auto 0;
}

.list:hover {
  background-color: #eee;
}
