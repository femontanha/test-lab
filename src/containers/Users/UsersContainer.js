import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchUsers } from '../../actions/users';
import Users from '../../components/Users/Users';

const propTypes = {
  users: PropTypes.object.isRequired,
};

class UsersContainer extends PureComponent {
  componentDidMount() {
    this.props.fetchUsers();
  }

  render() {
    const { users } = this.props;

    return (
      <Users users={ users } />
    );
  }
}

const mapStateToProps = state => {
  return {
    users: state.users,
  }
};

const mapDispatchToProps = {
  fetchUsers,
}

UsersContainer.propTypes = propTypes;

export default connect(mapStateToProps, mapDispatchToProps)(UsersContainer);
