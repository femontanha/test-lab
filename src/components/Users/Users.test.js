import React from 'react';
import Users from './Users';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

test('render users', () => {
  const users = {
    '0': {
      id: 0,
      username: 'username',
      name: 'name',
      email: 'email',
      phone: 'phone',
      website: 'website'
    },
    '1': {
      id: 1,
      username: 'username',
      name: 'name',
      email: 'email',
      phone: 'phone',
      website: 'website'
    },
  }

  const component = renderer.create(
    <MemoryRouter initialEntries={[ '/users' ]}>
      <Users users={ users } />
    </MemoryRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
