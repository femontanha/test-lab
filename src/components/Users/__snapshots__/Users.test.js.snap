// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render users 1`] = `
<div
  className={undefined}
>
  <a
    className="app-button"
    href="/"
    onClick={[Function]}
  >
    View Posts
  </a>
  <ul
    className={undefined}
  >
    <li>
      username
    </li>
    <li>
      name
    </li>
    <li>
      email
    </li>
    <li>
      phone
    </li>
    <li>
      website
    </li>
  </ul>
  <ul
    className={undefined}
  >
    <li>
      username
    </li>
    <li>
      name
    </li>
    <li>
      email
    </li>
    <li>
      phone
    </li>
    <li>
      website
    </li>
  </ul>
</div>
`;
